File: oa/models/token.go
package models

import "time"

// refresh token，由oa 秘钥签发，有效期长, 存储在token表
// app token, 由app 秘钥签发，有效期短, 不存储
// OverPerm 非oa应用获取oa数据的权限，由用户设定
type Token struct {
	BaseModel
	UserID    string    `json:"user_id" gorm:"index;type:varchar(32)" methods:"post,list" parse:"json"`
	User      *User     `json:"-"`
	AppID     string    `json:"app_id" gorm:"index;type:varchar(32)" methods:"post,list" parse:"json"`
	App       *App      `json:"-"`
	ExpiredAt time.Time `json:"expired_at" methods:"*post,*patch" parse:"json"`
	OverPerm  string    `json:"over_perm" methods:"*post,*patch" parse:"json"`
	Device    string    `json:"device" methods:"*post" parse:"json"`
}

File: oaweb/composables/api/token.ts
//
// Copyright (C) 2024 veypi <i@veypi.com>
// 2024-10-11 14:36:07
// Distributed under terms of the MIT license.
//

import webapi from "./webapi"
import * as models from "./models"
export interface TokenSaltOpts {
  username: string
  typ?: string
}
// keep
export function TokenSalt(json: TokenSaltOpts) {
  return webapi.Post<{ id: string, salt: string }>(`/token/salt`, { json })
}
export interface PostOpts {
  user_id: string
  token?: string
  salt?: string
  code?: string
  app_id?: string
  expired_at?: Date
  over_perm?: string
  device?: string
}

// keep
export function Post(json: PostOpts) {
  return webapi.Post<string>(`/token`, { json })
}
export function Get(token_id: string) {
  return webapi.Get<models.Token>(`/token/${token_id}`, {})
}

export interface PatchOpts {
  expired_at?: Date
  over_perm?: string
}
export function Patch(token_id: string, json: PatchOpts) {
  return webapi.Patch<models.Token>(`/token/${token_id}`, { json })
}

export function Delete(token_id: string) {
  return webapi.Delete<models.Token>(`/token/${token_id}`, {})
}

export interface ListOpts {
  user_id: string
  app_id: string
}
export function List(json: ListOpts) {
  return webapi.Get<models.Token>(`/token`, { json })
}

